src/titiler/extensions/titiler/extensions/wms.py

```diff
@@ -0,0 +1,534 @@
+                    )
+
+                crs = CRS.from_user_input(req.get("crs", req.get("srs")))
+
+                bbox = list(map(float, req["bbox"].split(",")))
+                if len(bbox) != 4:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Invalid 'BBOX' parameters: {req['bbox']}. Needs 4 coordinates separated by commas",
+                    )
+
+                if version == "1.3.0":
+                    # WMS 1.3.0 is lame and flips the coords of EPSG:4326
+                    # EPSG:4326 refers to WGS 84 geographic latitude, then longitude.
+                    # That is, in this CRS the x axis corresponds to latitude, and the y axis to longitude.
+                    if crs == CRS.from_epsg(4326):
+                        bbox = [
+                            bbox[1],
+                            bbox[0],
+                            bbox[3],
+                            bbox[2],
+                        ]
+
+                    # Overwrite CRS:84 with EPSG:4326 when specified
+                    # “CRS:84” refers to WGS 84 geographic longitude and latitude expressed in decimal degrees
+                    elif crs == CRS.from_user_input("CRS:84"):
+                        crs = CRS.from_epsg(4326)
+
+                if transparent := req.get("transparent", False):
+                    if str(transparent).lower() == "true":
+                        transparent = True
+
+                    elif str(transparent).lower() == "false":
+                        transparent = False
+
+                    else:
+                        raise HTTPException(
+                            status_code=400,
+                            detail=f"Invalid 'TRANSPARENT' parameter: {transparent}. Should be one of ['FALSE', 'TRUE'].",
+                        )
+
+                if req["format"] not in self.supported_format:
+                    raise HTTPException(
+                        status_code=400,
+                        detail=f"Invalid 'FORMAT' parameter: {req['format']}. Should be one of {self.supported_format}.",
+                    )
+                format = ImageType(WMSMediaType(req["format"]).name)
+
+                height, width = int(req["height"]), int(req["width"])
+
+                def _reader(src_path: str):
+                    with rasterio.Env(**env):
+                        with factory.reader(src_path, **reader_params) as src_dst:
+                            return src_dst.part(
+                                bbox,
+                                width=width,
+                                height=height,
+                                dst_crs=crs,
+                                bounds_crs=crs,
+                                **layer_params,
+                                **dataset_params,
+                            )
+
+                image, assets_used = mosaic_reader(
```

Comment:
In theory we should only be passing a single layer in the `layers` qs param, although I supposed this could work for a very simple mosaic method, absent MosaicJson. Is this the intended purpose rather than just getting the image out with .part() directly? How would this work with the MosaicTilerFactory if a user specified a custom backend?

Reply:
so here we use simple mosaic, meaning you just need to pass multiple file name, there is no idea of mosaic json or mosaic backend. 

> How would this work with the MosaicTilerFactory if a user specified a custom backend?

It won't work for mosaic backend because they don't implement `.part()` method (I might add it in titiler-pgstac)